### PR TITLE
feat(jolt-core): Adding MOVSIGN instruction

### PR DESCRIPTION
**Pull request by mw2000**
We now support the MOVSIGN instruction as part of the RISCV M extensions

### CODE REVIEW

File: jolt-core/src/jolt/instruction/movsign.rs [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+
+impl<const WORD_SIZE: usize> JoltInstruction for MOVSIGNInstruction<WORD_SIZE> {
+    fn operands(&self) -> (u64, u64) {
+        (self.0, 0)
+    }
+
+    fn combine_lookups<F: JoltField>(&self, vals: &[F], C: usize, _: usize) -> F {
+        let val = vals[0];
+        let mut result = F::zero();
+        for i in 0..C {
```
Comment (moodlezoup) [inline jolt-core/src/jolt/instruction/movsign.rs:33]: actually could we change this to: 
```suggestion
        for i in 0..WORD_SIZE / 16 {
```

File: jolt-core/src/jolt/instruction/movsign.rs [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+
+    use crate::{jolt::instruction::JoltInstruction, jolt_instruction_test};
+
+    use super::MOVSIGNInstruction;
+
+    #[test]
+    fn movsign_instruction_32_e2e() {
+        let mut rng = test_rng();
+        const C: usize = 2;
+        const M: usize = 1 << 16;
```
Comment (moodlezoup) [inline jolt-core/src/jolt/instruction/movsign.rs:97]: Once we make the above change, we should be able to change this to:
```suggestion
        const C: usize = 4;
        const M: usize = 1 << 16;
```

File: jolt-core/src/jolt/instruction/movsign.rs [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+            let instruction = MOVSIGNInstruction::<32>(x);
+            jolt_instruction_test!(instruction);
+        }
+    }
+
+    #[test]
+    fn movsign_instruction_64_e2e() {
+        let mut rng = test_rng();
+        const C: usize = 4;
+        const M: usize = 1 << 16;
```
Comment (moodlezoup) [inline jolt-core/src/jolt/instruction/movsign.rs:111]: And similarly: 
```suggestion
        const C: usize = 8;
        const M: usize = 1 << 16;
```

Comment (moodlezoup) [reply jolt-core/src/jolt/instruction/movsign.rs:111]: (for context, Jolt assumes elsewhere that `C * log2(M) = 2 * WORD_SIZE`)

**Review by moodlezoup — COMMENTED** [review]
Nice work, just a couple of nits

**Comment by mw2000** [comment]
@moodlezoup addressed all nits, ready for re-review

**Review by moodlezoup — APPROVED** [review]
Thanks!

**Pull request merged**